web/src/app/add-stuff/add-stuff.component.ts

```diff
@@ -35,12 +35,14 @@ export class AddStuffComponent implements OnInit {
 
 
   save() {
-    this.stuffService.addStuff(this.addStuff.value, this.getCategoryIds()).subscribe(
-      stuff => {
-        console.log(stuff);
-        this.router.navigate(['/stuff/' + stuff.id]);
-      }
-    );
+    setTimeout(() => {
```

Comment:
I got your point, but no, we won't do this. :) 
1. We need to wait 1 second even if save of a category will be in 1ms (1000 times longer waiting! :).
2. We still fail if save of a category will be longer than 1 second. 

Now, what we really need to do...
Our method has one problem - asynchronous call of the backend in the case when we want to persist category. This means we exit this method before it really finishes its work. And it's ok for all the cases except one - when we wanna save our "stuff". In this case, we need to wait while the request will be finished. 
So, we need to add a boolean property (for example, "isSaving") in the add-component. By default it's false and it will be set in the persist category method to true. In the subscriber of our request, we should set it back to false (you need to check you'll have it in case of success and error cases, I'll expect a test for these cases ). 
we should make our form "read-only" based on this property.